docs/src/index.md

```diff
@@ -0,0 +1,128 @@
+# LinearMaps.jl
+
+*A Julia package for defining and working with linear maps, also known as linear transformations or linear operators acting on vectors. The only requirement for a LinearMap is that it can act on a vector (by multiplication) efficiently.*
+
+## Installation
+
+`LinearMaps.jl` is a registered package and can be installed via
+
+    pkg> add LinearMaps
+
+in package mode, to be entered by typing `]` in the Julia REPL.
+
+## Examples
+
+Let
+
+    A = LinearMap(rand(10, 10))
+    B = LinearMap(cumsum, reverse∘cumsum∘reverse, 10)
+    
+be a matrix- and function-based linear map, respectively. Then the following code just works,
+indistinguishably from the case when `A` and `B` are both `AbstractMatrix`-typed objects.
+
+```
+3.0A + 2B
+A*B'
+[A B; B A]
+kron(A, B)
+```
+
+The `LinearMap` type and corresponding methods combine well with the following packages:
+* [Arpack.jl](https://github.com/JuliaLinearAlgebra/Arpack.jl): iterative eigensolver
+  `eigs` and SVD `svds`;
+* [IterativeSolvers.jl](https://github.com/JuliaMath/IterativeSolvers.jl): iterative
+  solvers, eigensolvers, and SVD;
+* [KrylovKit.jl](https://github.com/Jutho/KrylovKit.jl): Krylov-based algorithms for linear problems, singular value and eigenvalue problems
+* [TSVD.jl](https://github.com/andreasnoack/TSVD.jl): truncated SVD `tsvd`.
+
+```julia
+using LinearMaps
+import Arpack, IterativeSolvers, KrylovKit, TSVD
+
+# Example 1, 1-dimensional Laplacian with periodic boundary conditions
+function leftdiff!(y::AbstractVector, x::AbstractVector) # left difference assuming periodic boundary conditions
+    N = length(x)
+    length(y) == N || throw(DimensionMismatch())
+    @inbounds for i=1:N
+        y[i] = x[i] - x[mod1(i-1, N)]
+    end
+    return y
+end
+
+function mrightdiff!(y::AbstractVector, x::AbstractVector) # minus right difference
+    N = length(x)
+    length(y) == N || throw(DimensionMismatch())
+    @inbounds for i=1:N
+        y[i] = x[i] - x[mod1(i+1, N)]
+    end
+    return y
+end
+
+D = LinearMap(leftdiff!, mrightdiff!, 100; ismutating=true) # by default has eltype(D) = Float64
+
+Arpack.eigs(D'D; nev=3, which=:SR) # note that D'D is recognized as symmetric => real eigenfact
+Arpack.svds(D; nsv=3)
+
+Σ, L = IterativeSolvers.svdl(D; nsv=3)
+
+TSVD.tsvd(D, 3)
+
+# Example 2, 1-dimensional Laplacian
+A = LinearMap(100; issymmetric=true, ismutating=true) do C, B
+    C[1] = -2B[1] + B[2]
+    for i in 2:length(B)-1
+        C[i] = B[i-1] - 2B[i] + B[i+1]
+    end
+    C[end] = B[end-1] - 2B[end]
+    return C
+end
+
+Arpack.eigs(-A; nev=3, which=:SR)
+
+# Example 3, 2-dimensional Laplacian
+Δ = kronsum(A, A)
+
+Arpack.eigs(Δ; nev=3, which=:LR)
+KrylovKit.eigsolve(x -> Δ*x, size(Δ, 1), 3, :LR)
```

Comment:
With the previous comment in place, this could simply become `eigsolve(Δ, size(Δ, 1), 3, :LR)`